### PR TITLE
Documentation - Update Use Cases for Tag and Person

### PR DESCRIPTION
**Pull request by KwanHW**
# Changes
This PR updates the Use Cases for all Person and Tag functionalities. 

A live demo of the changes can be found [here](https://kwanhw.github.io/tp/DeveloperGuide.html).

# Related Issues
None.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -373,96 +373,249 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  * Use case resumes from step 2.<br>&nbsp;
 
-**Use case: Delete a person**
+* 2a. ContaX has no persons to list.
+  * Use case ends.<br>&nbsp;
 
-**MSS**
+* 4a. ContaX detects that a required person attribute was not supplied.
+    * 4a1. ContaX shows an error message indicating that there is a missing required attribute.
+    * Use case ends.<br>&nbsp;
```
Comment (sharpstorm) [inline docs/DeveloperGuide.md:439]: Delete does not require any attributes, just the index

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -373,96 +373,249 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+     
+* 4b. ContaX detects that a supplied attribute has an invalid value
+    * 4b1. ContaX shows an error message indicating that the supplied parameter is invalid.
+    * 4b2. ContaX shows the expected allowed values.
+    * Use case ends.<br>&nbsp;
 
-    * Use case resumes at step 2.
+* 5a. ContaX cannot find the requested person to delete.
+    * 5a1. ContaX shows an error message indicating that no such person exists.
+    * Use case ends.<br>&nbsp;
```
Comment (sharpstorm) [inline docs/DeveloperGuide.md:481]: If you want to say requested person to **edit** is not found, then you should place it at step 3.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -373,96 +373,249 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 5b. The person already exists.
+    * 5b1. ContaX shows an error message indicating that the person already exists.
+    * Use case ends.<br>&nbsp;
```
Comment (sharpstorm) [inline docs/DeveloperGuide.md:485]: I think `person with same name` would be a better wording.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -373,96 +373,249 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+5. ContaX shows that the tag has been deleted successfully.
+   Use case ends.
+
+**Extensions**
+
+* 3a. ContaX detects that the required tag attributed was not supplied.
+    * 3a1. ContaX shows an error message indicating that there is a missing required attribute.
+    * Use case ends.<br>&nbsp;
+
+* 4a. ContaX cannot find the requested tag to edit.
```
Comment (sharpstorm) [inline docs/DeveloperGuide.md:593]: It should be delete here, not edit.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -373,96 +373,249 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**UC3: Delete Person**
 
-    * 1b1. ContaX shows an error message indicating that the given parameter is of invalid format.
+**MSS**
 
-    * Use case ends.
+1. User requests to **list persons (UC1)**.
+2. ContaX displays a list of persons.
+3. User requests to delete a person.
+4. User enters details to delete person.
```
[the remark this answers is not in the document]
Comment (KwanHW) [reply docs/DeveloperGuide.md]: Fixed in https://github.com/AY2122S2-CS2103-W17-1/tp/pull/263/commits/b635bf79c7071c3d4393bb9cd86306ce5e7bfd5f

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (KwanHW) [reply docs/DeveloperGuide.md]: Fixed in https://github.com/AY2122S2-CS2103-W17-1/tp/pull/263/commits/b635bf79c7071c3d4393bb9cd86306ce5e7bfd5f

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (KwanHW) [reply docs/DeveloperGuide.md]: Fixed in https://github.com/AY2122S2-CS2103-W17-1/tp/pull/263/commits/7f066dcdd276c2f95aac11a8da54e000f9667dba

**Review by sharpstorm — APPROVED** [review]
LGTM.

**Review by dandaandaaaaaan — APPROVED** [review]
lgtm

**Pull request merged**